// react-native-table-optimize/lib/module/components/TableRow.js
import React from 'react';
import { StyleSheet, View, TouchableOpacity, Text } from 'react-native';
import * as THEME from './Theme';

const TableRow = _ref => {
  let {
    width = '100%',
    onHandleItemRow,
    textStyle = {},
    itemStyle = {},
    dataRow = [],
    dataSpacing = []
  } = _ref;
  return /*#__PURE__*/React.createElement(TouchableOpacity, {
    activeOpacity: 1,
    onPress: onHandleItemRow,
    style: { ...styles.container,
      width: width
    }
  }, (dataSpacing === null || dataSpacing === void 0 ? void 0 : dataSpacing.length) != 0 && dataSpacing.map((itemWidth, idx) => {
    return /*#__PURE__*/React.createElement(View, {
      key: idx,
      style: {
        width: itemWidth,
        ...itemStyle
      }
    }, /*#__PURE__*/React.createElement(Text, {
      style: {
        color: THEME.WHITE_COLOR,
        ...textStyle
      }
    }, // _.get(dataRow[idx], [keyDataRow]) ?  _.get(dataRow[idx], [keyDataRow]) : ''
    dataRow[idx] ? dataRow[idx] : ''));
  }));
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flexWrap: 'wrap'
  }
});
export default TableRow;
